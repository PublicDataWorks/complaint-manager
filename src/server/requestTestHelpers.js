import jwt from "jsonwebtoken";
import fs from "fs";
import path from "path";

const config = require("./config/config")[process.env.NODE_ENV];

function buildTokenWithPermissions(permissions, nickname) {
  const privateKeyPath = path.join(__dirname, "config", "test", "private.pem");
  const cert = fs.readFileSync(privateKeyPath);

  const payload = {
    foo: "bar",
    scope: `${config.authentication.scope} ${permissions}`
  };
  payload[`${config.authentication.nicknameKey}`] = nickname;

  const options = {
    audience: config.authentication.audience,
    issuer: config.authentication.issuer,
    algorithm: config.authentication.algorithm
  };

  return jwt.sign(payload, cert, options);
}

module.exports = buildTokenWithPermissions;
