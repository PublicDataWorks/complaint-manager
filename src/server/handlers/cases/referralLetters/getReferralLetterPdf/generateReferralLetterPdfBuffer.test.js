import ReferralLetter from "../../../../testHelpers/ReferralLetter";
import models from "../../../../policeDataManager/models";
import { cleanupDatabase } from "../../../../testHelpers/requestTestHelpers";
import timekeeper from "timekeeper";
import Case from "../../../../../sharedTestHelpers/case";
import {
  CASE_STATUS,
  COMPLAINANT,
  RANK_INITIATED
} from "../../../../../sharedUtilities/constants";
import generateReferralLetterPdfBuffer, {
  generateLetterPdfHtml
} from "./generateReferralLetterPdfBuffer";
import { generateReferralLetterBodyAndAuditDetails } from "../generateReferralLetterBodyAndAuditDetails";
import Officer from "../../../../../sharedTestHelpers/Officer";
import CaseOfficer from "../../../../../sharedTestHelpers/caseOfficer";
const {
  SENDER_NAME
} = require(`${process.env.REACT_APP_INSTANCE_FILES_DIR}/referralLetterDefaults`);

const AWS = require("aws-sdk");
jest.mock("aws-sdk");

jest.mock("html-pdf", () => ({
  create: (html, pdfOptions) => ({
    toBuffer: callback => {
      callback(null, html);
    }
  })
}));

jest.mock("../generateReferralLetterBodyAndAuditDetails", () => {
  return {
    generateReferralLetterBodyAndAuditDetails: jest.fn(
      (caseId, transaction) => {
        return {
          referralLetterData: {},
          auditDetails: {}
        };
      }
    )
  };
});

describe("generateReferralLetterPdfBuffer", function () {
  let existingCase, referralLetter, timeOfDownload, officer;

  afterEach(async () => {
    await cleanupDatabase();
    generateReferralLetterBodyAndAuditDetails.mockClear();
    timekeeper.reset();
  });

  afterAll(async () => {
    await models.sequelize.close();
  });

  beforeEach(async () => {
    timeOfDownload = new Date("2018-07-01 19:00:22 CDT");
    timekeeper.freeze(timeOfDownload);

    let s3 = AWS.S3.mockImplementation(() => ({
      config: {
        loadFromPath: jest.fn(),
        update: jest.fn()
      }
    }));

    const officerAttributes = new Officer.Builder()
      .defaultOfficer()
      .withId(undefined);

    officer = await models.officer.create(officerAttributes, {
      auditUser: "user"
    });

    const complainantOfficer = new CaseOfficer.Builder()
      .defaultCaseOfficer()
      .withId(undefined)
      .withOfficerId(officer.id)
      .withCreatedAt(new Date("2018-09-22"))
      .withRoleOnCase(COMPLAINANT);

    const caseAttributes = new Case.Builder()
      .defaultCase()
      .withId(12070)
      .withFirstContactDate("2017-12-25")
      .withIncidentDate("2016-01-01")
      .withComplaintType(RANK_INITIATED)
      .withComplainantOfficers([complainantOfficer]);
    existingCase = await models.cases.create(caseAttributes, {
      include: [
        {
          model: models.case_officer,
          as: "complainantOfficers",
          auditUser: "someone"
        }
      ],
      auditUser: "someone"
    });
    await existingCase.update(
      { status: CASE_STATUS.ACTIVE },
      { auditUser: "test" }
    );
    await existingCase.update(
      { status: CASE_STATUS.LETTER_IN_PROGRESS },
      { auditUser: "test" }
    );
  });

  describe("sender is 'sender address'", () => {
    beforeEach(async () => {
      const referralLetterAttributes = new ReferralLetter.Builder()
        .defaultReferralLetter()
        .withId(undefined)
        .withCaseId(existingCase.id)
        .withRecipient("recipient title and name")
        .withRecipientAddress("recipient address")
        .withSender("sender address")
        .withTranscribedBy("transcriber")
        .withIncludeRetaliationConcerns(true);

      referralLetter = await models.referral_letter.create(
        referralLetterAttributes,
        {
          auditUser: "test"
        }
      );
    });
    test("generates letter pdf html correctly", async () => {
      const letterBody = "<p> Letter Body </p>";
      const pdfData = {
        referralLetter: {
          recipient: "Recipient Title and Name",
          recipientAddress: "Recipient Address",
          sender: "Sender Address\n Sender Address Second Line",
          transcribedBy: "Transcriber"
        },
        caseReference: "CC-2011-0099"
      };

      const letterPdfHtml = await generateLetterPdfHtml(
        letterBody,
        pdfData,
        false
      );
      expect(letterPdfHtml).toMatchSnapshot();
    });

    test("should update case reference prefix to AC when primary complainant is anonymized", async () => {
      await models.case_officer.update(
        {
          isAnonymous: true
        },
        {
          where: {
            officerId: officer.id
          },
          auditUser: "test user"
        }
      );
      await models.sequelize.transaction(async transaction => {
        const pdfResultsAndAuditDetails = await generateReferralLetterPdfBuffer(
          existingCase.id,
          true,
          transaction
        );

        expect(pdfResultsAndAuditDetails.pdfBuffer).toMatchSnapshot();
      });
    });

    test("pdf create gets called with expected letter html when letter is generated", async () => {
      await models.sequelize.transaction(async transaction => {
        const pdfResultsAndAuditDetails = await generateReferralLetterPdfBuffer(
          existingCase.id,
          true,
          transaction
        );
        expect(pdfResultsAndAuditDetails.pdfBuffer).toMatchSnapshot();
      });
    });

    test("pdf create gets called with expected letter html when letter is edited", async () => {
      await referralLetter.update(
        { editedLetterHtml: "Custom Letter HTML" },
        { auditUser: "someone" }
      );

      await models.sequelize.transaction(async transaction => {
        const pdfResultsAndAuditDetails = await generateReferralLetterPdfBuffer(
          existingCase.id,
          true,
          transaction
        );
        expect(pdfResultsAndAuditDetails.pdfBuffer).toMatchSnapshot();
      });
    });

    test("signature is not included when sender is not official sender even if includeSignature is true", async () => {
      await referralLetter.update(
        { editedLetterHtml: "Custom Letter HTML" },
        { auditUser: "someone" }
      );
      await models.sequelize.transaction(async transaction => {
        const pdfResultsAndAuditDetails = await generateReferralLetterPdfBuffer(
          existingCase.id,
          true,
          transaction
        );
        expect(pdfResultsAndAuditDetails.pdfBuffer).toMatchSnapshot();
      });
    });

    test("unedited letter generates pdf from case data", async () => {
      await models.sequelize.transaction(async transaction => {
        await generateReferralLetterPdfBuffer(
          existingCase.id,
          false,
          transaction
        );
        expect(generateReferralLetterBodyAndAuditDetails).toHaveBeenCalledWith(
          existingCase.id,
          transaction
        );
      });
    });

    test("edited letter generates pdf from saved data", async () => {
      await referralLetter.update(
        {
          editedLetterHtml: "<p> edited </p>"
        },
        { auditUser: "test" }
      );
      await models.sequelize.transaction(async transaction => {
        await generateReferralLetterPdfBuffer(
          existingCase.id,
          false,
          transaction
        );
      });
      expect(generateReferralLetterBodyAndAuditDetails).not.toHaveBeenCalled();
    });
  });

  describe("sender is official sender", () => {
    beforeEach(async () => {
      const referralLetterAttributes = new ReferralLetter.Builder()
        .defaultReferralLetter()
        .withId(undefined)
        .withCaseId(existingCase.id)
        .withRecipient("recipient title and name")
        .withRecipientAddress("recipient address")
        .withSender(SENDER_NAME)
        .withTranscribedBy("transcriber")
        .withIncludeRetaliationConcerns(true);

      referralLetter = await models.referral_letter.create(
        referralLetterAttributes,
        {
          auditUser: "test"
        }
      );
    });

    test("signature is included when sender is official sender and includeSignature is true", async () => {
      await models.sequelize.transaction(async transaction => {
        const pdfResultsAndAuditDetails = await generateReferralLetterPdfBuffer(
          existingCase.id,
          true,
          transaction
        );
        expect(pdfResultsAndAuditDetails.pdfBuffer).toMatchSnapshot();
      });
    });

    test("signature is not included when includeSignature is false", async () => {
      await models.sequelize.transaction(async transaction => {
        const pdfResultsAndAuditDetails = await generateReferralLetterPdfBuffer(
          existingCase.id,
          false,
          transaction
        );
        expect(pdfResultsAndAuditDetails.pdfBuffer).toMatchSnapshot();
      });
    });
  });
});
