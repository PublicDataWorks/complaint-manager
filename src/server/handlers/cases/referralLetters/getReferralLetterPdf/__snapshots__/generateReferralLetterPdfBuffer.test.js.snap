// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateReferralLetterPdfBuffer sender is 'sender address' generates letter pdf html correctly 1`] = `
"<html>
  <head>
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"pdf.css\\">
  </head>
  <body>
    <div id=\\"pageHeader-first\\">
      <div style=\\"text-align: center;\\">
        <img style=\\"max-width: 223px\\" src=https://noipm-private-images.s3.amazonaws.com/header_text.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=a3b5589f1b637005ca37439b628dbd4e063280f4835bcfd1ce9b690fec9a09ab&X-Amz-SignedHeaders=host />
      </div>
      <div style=\\"text-align: center; margin-top: 8px\\">
        <img style=\\"max-width: 42px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
      </div>
    </div>
    <div id=\\"pageHeader\\" style=\\"font-size:8.5pt; color: #7F7F7F;\\">
      Recipient Title and Name<br/>
      July 01, 2018<br/>
      Page {{page}}
    </div>

    <div id=\\"pageFooter\\" style=\\"text-align: center; margin-top: 16px\\">
            <span  style=\\"display:inline-block; margin: 6px 16px 0 0\\">
                <img style=\\"max-width: 30px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
            </span>
      <span style=\\"display:inline-block; font-size:7pt; color: #7F7F7F;\\">
                INDEPENDENT POLICE MONITOR <br />
                2714 Canal Street, Suite 201 | NEW ORLEANS, LOUISIANA | 70119 <br />
                Phone (504) 309-9799| Fax (504) 309-7345
            </span>
      <span  style=\\"display:inline-block; width: 46px\\">&nbsp;</span>
    </div>
    <p style=\\"color: #7F7F7F;\\">
      SUSAN HUTSON
      <br>
      INDEPENDENT POLICE MONITOR
    </p>
    <p><br></p>
    July 01, 2018
    <p><br></p>
    Recipient Title and Name</br>
    Recipient Address
    <p><br></p>
    <p><strong>RE: Complaint Referral; OIPM Complaint CC-2011-0099</strong></p>
    <p><br></p>
    <p>Dear Recipient Title and Name:</p>
    <p><br></p>
    <p> Letter Body </p>
    <p><br></p>
    Sincerely,
    <p><br></p>
    <p><br></p>
    <p><br></p>
    Sender Address<br> Sender Address Second Line
    <p><br></p>
    Transcribed by: Transcriber
  </body>
</html>"
`;

exports[`generateReferralLetterPdfBuffer sender is 'sender address' pdf create gets called with expected letter html when letter is edited 1`] = `
"<html>
  <head>
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"pdf.css\\">
  </head>
  <body>
    <div id=\\"pageHeader-first\\">
      <div style=\\"text-align: center;\\">
        <img style=\\"max-width: 223px\\" src=https://noipm-private-images.s3.amazonaws.com/header_text.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=a3b5589f1b637005ca37439b628dbd4e063280f4835bcfd1ce9b690fec9a09ab&X-Amz-SignedHeaders=host />
      </div>
      <div style=\\"text-align: center; margin-top: 8px\\">
        <img style=\\"max-width: 42px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
      </div>
    </div>
    <div id=\\"pageHeader\\" style=\\"font-size:8.5pt; color: #7F7F7F;\\">
      recipient title and name<br/>
      July 01, 2018<br/>
      Page {{page}}
    </div>

    <div id=\\"pageFooter\\" style=\\"text-align: center; margin-top: 16px\\">
            <span  style=\\"display:inline-block; margin: 6px 16px 0 0\\">
                <img style=\\"max-width: 30px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
            </span>
      <span style=\\"display:inline-block; font-size:7pt; color: #7F7F7F;\\">
                INDEPENDENT POLICE MONITOR <br />
                2714 Canal Street, Suite 201 | NEW ORLEANS, LOUISIANA | 70119 <br />
                Phone (504) 309-9799| Fax (504) 309-7345
            </span>
      <span  style=\\"display:inline-block; width: 46px\\">&nbsp;</span>
    </div>
    <p style=\\"color: #7F7F7F;\\">
      SUSAN HUTSON
      <br>
      INDEPENDENT POLICE MONITOR
    </p>
    <p><br></p>
    July 01, 2018
    <p><br></p>
    recipient title and name</br>
    recipient address
    <p><br></p>
    <p><strong>RE: Complaint Referral; OIPM Complaint PO2017-0001</strong></p>
    <p><br></p>
    <p>Dear recipient title and name:</p>
    <p><br></p>
    Custom Letter HTML
    <p><br></p>
    Sincerely,
    <p><br></p>
    <p><br></p>
    <p><br></p>
    sender address
    <p><br></p>
    Transcribed by: transcriber
  </body>
</html>"
`;

exports[`generateReferralLetterPdfBuffer sender is 'sender address' pdf create gets called with expected letter html when letter is generated 1`] = `
"<html>
  <head>
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"pdf.css\\">
  </head>
  <body>
    <div id=\\"pageHeader-first\\">
      <div style=\\"text-align: center;\\">
        <img style=\\"max-width: 223px\\" src=https://noipm-private-images.s3.amazonaws.com/header_text.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=a3b5589f1b637005ca37439b628dbd4e063280f4835bcfd1ce9b690fec9a09ab&X-Amz-SignedHeaders=host />
      </div>
      <div style=\\"text-align: center; margin-top: 8px\\">
        <img style=\\"max-width: 42px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
      </div>
    </div>
    <div id=\\"pageHeader\\" style=\\"font-size:8.5pt; color: #7F7F7F;\\">
      recipient title and name<br/>
      July 01, 2018<br/>
      Page {{page}}
    </div>

    <div id=\\"pageFooter\\" style=\\"text-align: center; margin-top: 16px\\">
            <span  style=\\"display:inline-block; margin: 6px 16px 0 0\\">
                <img style=\\"max-width: 30px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
            </span>
      <span style=\\"display:inline-block; font-size:7pt; color: #7F7F7F;\\">
                INDEPENDENT POLICE MONITOR <br />
                2714 Canal Street, Suite 201 | NEW ORLEANS, LOUISIANA | 70119 <br />
                Phone (504) 309-9799| Fax (504) 309-7345
            </span>
      <span  style=\\"display:inline-block; width: 46px\\">&nbsp;</span>
    </div>
    <p style=\\"color: #7F7F7F;\\">
      SUSAN HUTSON
      <br>
      INDEPENDENT POLICE MONITOR
    </p>
    <p><br></p>
    July 01, 2018
    <p><br></p>
    recipient title and name</br>
    recipient address
    <p><br></p>
    <p><strong>RE: Complaint Referral; OIPM Complaint PO2017-0001</strong></p>
    <p><br></p>
    <p>Dear recipient title and name:</p>
    <p><br></p>
    
    <p><br></p>
    Sincerely,
    <p><br></p>
    <p><br></p>
    <p><br></p>
    sender address
    <p><br></p>
    Transcribed by: transcriber
  </body>
</html>"
`;

exports[`generateReferralLetterPdfBuffer sender is 'sender address' should update case reference prefix to AC when primary complainant is anonymized 1`] = `
"<html>
  <head>
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"pdf.css\\">
  </head>
  <body>
    <div id=\\"pageHeader-first\\">
      <div style=\\"text-align: center;\\">
        <img style=\\"max-width: 223px\\" src=https://noipm-private-images.s3.amazonaws.com/header_text.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=a3b5589f1b637005ca37439b628dbd4e063280f4835bcfd1ce9b690fec9a09ab&X-Amz-SignedHeaders=host />
      </div>
      <div style=\\"text-align: center; margin-top: 8px\\">
        <img style=\\"max-width: 42px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
      </div>
    </div>
    <div id=\\"pageHeader\\" style=\\"font-size:8.5pt; color: #7F7F7F;\\">
      recipient title and name<br/>
      July 01, 2018<br/>
      Page {{page}}
    </div>

    <div id=\\"pageFooter\\" style=\\"text-align: center; margin-top: 16px\\">
            <span  style=\\"display:inline-block; margin: 6px 16px 0 0\\">
                <img style=\\"max-width: 30px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
            </span>
      <span style=\\"display:inline-block; font-size:7pt; color: #7F7F7F;\\">
                INDEPENDENT POLICE MONITOR <br />
                2714 Canal Street, Suite 201 | NEW ORLEANS, LOUISIANA | 70119 <br />
                Phone (504) 309-9799| Fax (504) 309-7345
            </span>
      <span  style=\\"display:inline-block; width: 46px\\">&nbsp;</span>
    </div>
    <p style=\\"color: #7F7F7F;\\">
      SUSAN HUTSON
      <br>
      INDEPENDENT POLICE MONITOR
    </p>
    <p><br></p>
    July 01, 2018
    <p><br></p>
    recipient title and name</br>
    recipient address
    <p><br></p>
    <p><strong>RE: Complaint Referral; OIPM Complaint AC2017-0001</strong></p>
    <p><br></p>
    <p>Dear recipient title and name:</p>
    <p><br></p>
    
    <p><br></p>
    Sincerely,
    <p><br></p>
    <p><br></p>
    <p><br></p>
    sender address
    <p><br></p>
    Transcribed by: transcriber
  </body>
</html>"
`;

exports[`generateReferralLetterPdfBuffer sender is 'sender address' signature is not included when sender is not stella even if includeSignature is true 1`] = `
"<html>
  <head>
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"pdf.css\\">
  </head>
  <body>
    <div id=\\"pageHeader-first\\">
      <div style=\\"text-align: center;\\">
        <img style=\\"max-width: 223px\\" src=https://noipm-private-images.s3.amazonaws.com/header_text.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=a3b5589f1b637005ca37439b628dbd4e063280f4835bcfd1ce9b690fec9a09ab&X-Amz-SignedHeaders=host />
      </div>
      <div style=\\"text-align: center; margin-top: 8px\\">
        <img style=\\"max-width: 42px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
      </div>
    </div>
    <div id=\\"pageHeader\\" style=\\"font-size:8.5pt; color: #7F7F7F;\\">
      recipient title and name<br/>
      July 01, 2018<br/>
      Page {{page}}
    </div>

    <div id=\\"pageFooter\\" style=\\"text-align: center; margin-top: 16px\\">
            <span  style=\\"display:inline-block; margin: 6px 16px 0 0\\">
                <img style=\\"max-width: 30px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
            </span>
      <span style=\\"display:inline-block; font-size:7pt; color: #7F7F7F;\\">
                INDEPENDENT POLICE MONITOR <br />
                2714 Canal Street, Suite 201 | NEW ORLEANS, LOUISIANA | 70119 <br />
                Phone (504) 309-9799| Fax (504) 309-7345
            </span>
      <span  style=\\"display:inline-block; width: 46px\\">&nbsp;</span>
    </div>
    <p style=\\"color: #7F7F7F;\\">
      SUSAN HUTSON
      <br>
      INDEPENDENT POLICE MONITOR
    </p>
    <p><br></p>
    July 01, 2018
    <p><br></p>
    recipient title and name</br>
    recipient address
    <p><br></p>
    <p><strong>RE: Complaint Referral; OIPM Complaint PO2017-0001</strong></p>
    <p><br></p>
    <p>Dear recipient title and name:</p>
    <p><br></p>
    Custom Letter HTML
    <p><br></p>
    Sincerely,
    <p><br></p>
    <p><br></p>
    <p><br></p>
    sender address
    <p><br></p>
    Transcribed by: transcriber
  </body>
</html>"
`;

exports[`generateReferralLetterPdfBuffer sender is stella signature is included when sender is stella and includeSignature is true 1`] = `
"<html>
  <head>
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"pdf.css\\">
  </head>
  <body>
    <div id=\\"pageHeader-first\\">
      <div style=\\"text-align: center;\\">
        <img style=\\"max-width: 223px\\" src=https://noipm-private-images.s3.amazonaws.com/header_text.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=a3b5589f1b637005ca37439b628dbd4e063280f4835bcfd1ce9b690fec9a09ab&X-Amz-SignedHeaders=host />
      </div>
      <div style=\\"text-align: center; margin-top: 8px\\">
        <img style=\\"max-width: 42px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
      </div>
    </div>
    <div id=\\"pageHeader\\" style=\\"font-size:8.5pt; color: #7F7F7F;\\">
      recipient title and name<br/>
      July 01, 2018<br/>
      Page {{page}}
    </div>

    <div id=\\"pageFooter\\" style=\\"text-align: center; margin-top: 16px\\">
            <span  style=\\"display:inline-block; margin: 6px 16px 0 0\\">
                <img style=\\"max-width: 30px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
            </span>
      <span style=\\"display:inline-block; font-size:7pt; color: #7F7F7F;\\">
                INDEPENDENT POLICE MONITOR <br />
                2714 Canal Street, Suite 201 | NEW ORLEANS, LOUISIANA | 70119 <br />
                Phone (504) 309-9799| Fax (504) 309-7345
            </span>
      <span  style=\\"display:inline-block; width: 46px\\">&nbsp;</span>
    </div>
    <p style=\\"color: #7F7F7F;\\">
      SUSAN HUTSON
      <br>
      INDEPENDENT POLICE MONITOR
    </p>
    <p><br></p>
    July 01, 2018
    <p><br></p>
    recipient title and name</br>
    recipient address
    <p><br></p>
    <p><strong>RE: Complaint Referral; OIPM Complaint PO2017-0001</strong></p>
    <p><br></p>
    <p>Dear recipient title and name:</p>
    <p><br></p>
    
    <p><br></p>
    Sincerely,
    <p><br></p>
    <img style=\\"max-height: 55px\\" src=https://noipm-private-images.s3.amazonaws.com/signature.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=79f47086619c1e04f7df7ce165d0fd46d2fa3e1b43b925f70545af43c8bed357&X-Amz-SignedHeaders=host />
    <p><br></p>
    Stella Cziment
    <p><br></p>
    Transcribed by: transcriber
  </body>
</html>"
`;

exports[`generateReferralLetterPdfBuffer sender is stella signature is not included when includeSignature is false 1`] = `
"<html>
  <head>
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"pdf.css\\">
  </head>
  <body>
    <div id=\\"pageHeader-first\\">
      <div style=\\"text-align: center;\\">
        <img style=\\"max-width: 223px\\" src=https://noipm-private-images.s3.amazonaws.com/header_text.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=a3b5589f1b637005ca37439b628dbd4e063280f4835bcfd1ce9b690fec9a09ab&X-Amz-SignedHeaders=host />
      </div>
      <div style=\\"text-align: center; margin-top: 8px\\">
        <img style=\\"max-width: 42px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
      </div>
    </div>
    <div id=\\"pageHeader\\" style=\\"font-size:8.5pt; color: #7F7F7F;\\">
      recipient title and name<br/>
      July 01, 2018<br/>
      Page {{page}}
    </div>

    <div id=\\"pageFooter\\" style=\\"text-align: center; margin-top: 16px\\">
            <span  style=\\"display:inline-block; margin: 6px 16px 0 0\\">
                <img style=\\"max-width: 30px\\" src=https://noipm-private-images.s3.amazonaws.com/icon.ico?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20180702%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180702T000022Z&X-Amz-Expires=60&X-Amz-Signature=0fbad4e872a737ac62eb928bfeabf7413b6d6e4229bb7564544efe1404c5cf94&X-Amz-SignedHeaders=host />
            </span>
      <span style=\\"display:inline-block; font-size:7pt; color: #7F7F7F;\\">
                INDEPENDENT POLICE MONITOR <br />
                2714 Canal Street, Suite 201 | NEW ORLEANS, LOUISIANA | 70119 <br />
                Phone (504) 309-9799| Fax (504) 309-7345
            </span>
      <span  style=\\"display:inline-block; width: 46px\\">&nbsp;</span>
    </div>
    <p style=\\"color: #7F7F7F;\\">
      SUSAN HUTSON
      <br>
      INDEPENDENT POLICE MONITOR
    </p>
    <p><br></p>
    July 01, 2018
    <p><br></p>
    recipient title and name</br>
    recipient address
    <p><br></p>
    <p><strong>RE: Complaint Referral; OIPM Complaint PO2017-0001</strong></p>
    <p><br></p>
    <p>Dear recipient title and name:</p>
    <p><br></p>
    
    <p><br></p>
    Sincerely,
    <p><br></p>
    <p><br></p>
    <p><br></p>
    Stella Cziment
    <p><br></p>
    Transcribed by: transcriber
  </body>
</html>"
`;
